admin-base/ui.apps/src/main/js/apiImpl.js

```diff
@@ -471,7 +471,36 @@ class PerAdminImpl {
 
   populateReferencedBy(path) {
     return fetch('/admin/refBy.json' + path)
-        .then((data) => populateView('/state', 'referencedBy', data))
+        .then((data) => {
+          var refBy = data.referencedBy;        
+          for (var i = 0; i < refBy.length; ++i) {
+            //checks if object in position of array is valid
+            if (refBy[i].path){
+              //if count is not set, sets it
+              if (!refBy[i].count){
+                refBy[i].count = 0;
+              }
+              //will give it a count of 1, then if its the different element on the same page, will add again
+              refBy[i].count += 1;
+            }
+            //checks if there is an element next to current element
```

Comment:
@smcgrath0, consider this: what happens if the response from '/perapi/admin/refBy.json' does not include the `referencedBy.[]` elements sorted by path? Will this code still work?

Reply:
@smcgrath0 and @dna113p - Why can't we use a `map` and make this dead simple? I am no ES6 expert, but here's what I am thinking. 

```
let referencedBy = [
    {
        "name": "carousel",
        "path": "/content/themecleanflex/pages/library/carousel",
        "propertyName": "imagepath",
        "propertyPath": "/content/themecleanflex/pages/library/carousel/jcr:content/container22/carousel9/slides/c6",
    },
    {
        "name": "sample-owl",
        "path": "/content/themecleanflex/pages/library/carousel/sample-owl",
        "propertyName": "imagepath",
        "propertyPath": "/content/themecleanflex/pages/library/carousel/sample-owl/jcr:content/container175/carousel6/slides/c3",
    },
    {
        "name": "sample-owl",
        "path": "/content/themecleanflex/pages/library/carousel/sample-owl",
        "propertyName": "imagepath",
        "propertyPath": "/content/themecleanflex/pages/library/carousel/sample-owl/jcr:content/container175/carousel12/slides/c9",
    },
]

let result = referencedBy.reduce(
    (map => (r, a) => (!map.has(a.path) && map.set(a.path, r[r.push({ name: a.name, path: a.path, propertyName: a.propertyName, propertyPath: a.propertyPath, count: 0 }) - 1]), map.get(a.path).count++, r))(new Map),
    []
)

console.log(result)
```

It produces this:
```
[
  {
    name: 'carousel',
    path: '/content/themecleanflex/pages/library/carousel',
    propertyName: 'imagepath',
    propertyPath: '/content/themecleanflex/pages/library/carousel/jcr:content/container22/carousel9/slides/c6',
    count: 1
  },
  {
    name: 'sample-owl',
    path: '/content/themecleanflex/pages/library/carousel/sample-owl',
    propertyName: 'imagepath',
    propertyPath: '/content/themecleanflex/pages/library/carousel/sample-owl/jcr:content/container175/carousel6/slides/c3',
    count: 2
  }
]
```

Please let me know if I am way off.